검증 카카오 로그인 서비스 테스트 위해 일시적으로 회원가입시 password와 role USER 부분 저장되도록 로직 추가

denmark-dangnagui opened the pull request.

denmark-dangnagui committed:
검증 카카오 로그인 서비스 테스트 위해 일시적으로 회원가입시 password와 role USER 부분 저장되도록 로직 추가

somefood commented on src/main/java/main/sulsul/oauth/application/OAuthLoginService.java:
```diff
@@ -36,6 +37,8 @@ private Long newMember(OAuthInfoResponse oAuthInfoResponse) {
         Member member = Member.builder()
                 .email(oAuthInfoResponse.getEmail())
                 .username(oAuthInfoResponse.getNickname())
+                .role(Role.USER)
+                .password("1111")
```
작성하신 내용 잘 보았습니다.
role은 그대로 두셔도 될거 같은데,
password를 PasswordEncoder의 encode 메서드를 적용해주시면 좋겠습니다!

SecurityConfig에 BCryptPasswordEncoder 구현체를 빈으로 등록했으니, 주입받아 사용하시면 될겁니다~!

denmark-dangnagui replied:
시큐리티를 나중에 손봐야지 생각했었는데 주입받아서 쓰면 이미 있었군요 ㅎㅎ 수정하여 올리도록 하겠습니다!

denmark-dangnagui committed:
검증 리다이렉트 uri 변경 로직 추가 + passwordencoder 로직 추가 기입

somefood commented on src/main/java/main/sulsul/oauth/application/OAuthLoginService.java:
```diff
@@ -19,6 +21,9 @@ public class OAuthLoginService {
     private final MemberRepository memberRepository;
     private final AuthTokensGenerator authTokensGenerator;
     private final RequestOAuthInfoService requestOAuthInfoService;
+    public final PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
```
새로 메서드로 생성할 필요없이 Security Config에 있는걸 주입 받는 걸로 해보시면 좋겠습니다!

denmark-dangnagui replied:
아 SecurityConfig 자체를 주입받으면 될까요?
해보고 헷갈리면 말씀드리겠습니다!

somefood replied:
```java
@Slf4j
@Configuration
@EnableWebSecurity
public class SecurityConfig {

    @Bean
    public PasswordEncoder passwordEncoder() {
        return new BCryptPasswordEncoder();
    }
```

위 SecurityConfig 보시면 제가 빈으로 등록해둔게 있는데, 이 빈을 주입 받으란 거였습니다 ㅎㅎ

somefood commented on src/main/java/main/sulsul/oauth/application/OAuthLoginService.java:
```diff
@@ -38,8 +43,9 @@ private Long newMember(OAuthInfoResponse oAuthInfoResponse) {
                 .username(oAuthInfoResponse.getNickname())
                 .role(Role.USER)
-                .password("1111")
+                .password(passwordEncoder().encode("1111"))
```
1111 같은 매직 넘버 대신 상수로 빼 보시는게 어떨까요~?

denmark-dangnagui replied:
상수로 빼라는게 혹시 어떤거를 말씀하시는걸까요??

somefood replied:
1111을 final static 변수로 선언해서 의미있는 변수명을 가져보게 하는 것입니다!
단순 문자열만 넣는것보단 더 이해하기가 수월해 질겁니다!

아래 링크도 같이 참조해주세요!
https://velog.io/@jeongbeom4693/JAVA-%EC%83%81%EC%88%98-%EB%A7%A4%EC%A7%81-%EB%84%98%EB%B2%84Magic-Number

somefood commented on src/main/java/main/sulsul/oauth/domain/kakao/KakaoApiClient.java:
```diff
@@ -51,7 +51,8 @@ public String requestAccessToken(OAuthLoginParams params) {
         body.add("client_id", clientId);
         body.add("client_secret", clientSecret);
-        body.add("redirect_uri", "http://localhost:8080/login/oauth2/code/kakao");
+//        body.add("redirect_uri", "http://localhost:8080/login/oauth2/code/kakao"); //로컬
+        body.add("redirect_uri", "http://ec2-52-78-29-203.ap-northeast-2.compute.amazonaws.com:9090/login/oauth2/code/kakao");
```
추후에 application 프로필을 활용해서 로컬, 검증, 운영 환경 별로 URL 입력받을 수 있게 해주시면 좋겠습니다!

denmark-dangnagui replied:
넵! EC2에 올렸을 때 API 통신이 제대로 되는지 확인하고자 임시적으로 진행하였습니다! 수정해놓을게요!

denmark-dangnagui committed:
로그인 고도화
1. 로그인시 accessToken 만료되지 않는다면 정상적으로 넘어가도록 수정
2. 현재 accessToken 만료시 refershToken 값이 DB 값과 같다면 accessToken 값을 클라쪽에 반환하는 로직 개발중

denmark-dangnagui committed:
로그인 고도화
1. 정은님 카카오 개바자 도구 아이디로 변경

denmark-dangnagui committed:
로그인 고도화
redirect_uri 변경

denmark-dangnagui committed:
로그인 고도화
카카오디벨로퍼 시크릿과 클라이언트ID 테스트

denmark-dangnagui committed:
로그인 고도화
카카오디벨로퍼 시크릿과 클라이언트ID 테스트 - 2

denmark-dangnagui committed:
로그인 고도화
카카오디벨로퍼 시크릿과 클라이언트ID 테스트 - 3

denmark-dangnagui committed:
로그인 고도화
카카오디벨로퍼 시크릿과 클라이언트ID 테스트 - 4
클라이언트 id 값을 rest값에서 네이티브값으로 변경

denmark-dangnagui committed:
검증에서 로그인 테스트를 위해.

denmark-dangnagui committed:
로그인 고도화
- 토큰값 기준으로,
   1. accessToken이 정상토큰일 때
   2. accessToekn이 만료되었을 때 + refresh는 만료 x
     --> accessToken값을 반환 (+ 600 코드값과 함께)
   3. refreshToken이 만료되었을 때
     --> 만료되었다는 601 코드 값만 보내줌.
- 회원 조회시 MemberDto 로 조회되도록 + 컨트롤러단이 아닌 서비스 단에서 해결할 수 있도록 조정
- 회원가입 된 멤버 조회시 토큰 값이 없는 경우, 유효하지 않는 회원처리

denmark-dangnagui committed:
로그인 고도화
- 인가코드가 아닌 액세스토큰 값으로 로그인 진행

denmark-dangnagui committed:
로그인 고도화
- 인가코드가 아닌 액세스토큰 값으로 로그인 진행
- redirectUri 변경

denmark-dangnagui committed:
로그인 고도화
- 로그인 api uri 변경
/test -> /login 변경
